experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/client/Client.scala

```diff
@@ -162,44 +168,118 @@ class Client(configuration:AppConfig, yarnConf: YarnConfiguration, yarnClient: Y
     })
   }
 
-  def monitorAM(appContext: ApplicationSubmissionContext): Unit = {
-    val appId = appContext.getApplicationId
+  def delay(appId: ApplicationId): (ApplicationReport, YarnApplicationState) = {
+    Thread.sleep(1000)
+    val appReport = yarnClient.getApplicationReport(appId)
+    val appState = appReport.getYarnApplicationState
+    (appReport, appState)
+  }
+
+  def monitor(appId: ApplicationId): Unit = {
     var appReport = yarnClient.getApplicationReport(appId)
     var appState = appReport.getYarnApplicationState
-    while (appState != YarnApplicationState.FINISHED &&
-      appState != YarnApplicationState.KILLED &&
-      appState != YarnApplicationState.FAILED) {
-      Thread.sleep(1000)
-      appReport = yarnClient.getApplicationReport(appId)
-      appState = appReport.getYarnApplicationState
+    val monitor = getEnv("monitor").toBoolean
+    var watch = true
```

Comment:
if -monitor is provided on the command line then the client will poll the appmaster state until a terminal state is reached (FINISHED, KILLED, FAILED). Otherwise the client will exit when a non-terminal state of RUNNING is reached.